### PR TITLE
Add auction list displayer component and disable it on api

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Reasoning behind the pull request
- We use the same component `AuctionListSelector` for both computation at the end of the epoch and api calls simulation
- Since this component used to print a lot of data to the log, therefore causing increased storage needs, we needed to disable it for api calls
  
## Proposed changes
- Moved all storage expensive table displayers into a new component: `auctionListDisplayer`.
- This above mentioned displayer is used only for epoch start computation. For api component, we use a disabled one

## Testing procedure
- Run a system test with debug level, multiKey and enough nodes in network to always have the auction list filled
- After epoch 6 (when whole stakingV4 chain is enabled), we should periodically call `validator/auction` route and make sure that no table is printed
- Meanwhile, starting epoch 6, at each epoch change, we should always see the tables (  **3 tables**) ,    similar to this:

1.
```
DEBUG[2024-02-01 18:49:36.194] [..hStart/metachain] [metachain/1269/18448096/(BLOCK)] Initial nodes config in auction list
+------------------------------------------------------------------+------------------+------------------+-------------------+---------------+-----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Owner                                                            | Num staked nodes | Num active nodes | Num auction nodes | Total top up  | Top up per node | Auction list nodes                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
+------------------------------------------------------------------+------------------+------------------+-------------------+---------------+-----------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 0000000000000000000100000000000000000000000000000000000053ffffff | 2                | 1                | 1                 | 817.01201     | 408.50600       | 2b8e414f0d...195434e787                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
| f4b8d1aed9c04baa08b750203edadcee1bc3d963d66ce7000f5863a3fcd9f290 | 1                | 0                | 1                 | 0.00000       | 0.00000         | 752f4d283f...20371be80f                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |
| 000000000000000000010000000000000000000000000000000000007fffffff | 10               | 9                | 1                 | 31075.81928   | 3107.58192      | 22bd559821...d733faff89                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |

```

2.

```
DEBUG[2024-02-01 18:49:36.196] [..hStart/metachain] [metachain/1269/18448096/(BLOCK)] Selected nodes config from auction list
+------------------------------------------------------------------+------------------+----------------+---------------+-------------------+-----------------------------+------------------+---------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Owner                                                            | Num staked nodes | TopUp per node | Total top up  | Num auction nodes | Num qualified auction nodes | Num active nodes | Qualified top up per node | Selected auction list nodes                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
+------------------------------------------------------------------+------------------+----------------+---------------+-------------------+-----------------------------+------------------+---------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 0000000000000000000100000000000000000000000000000000000006ffffff | 50               | 2799.00877     | 139950.43854  | 4                 | 4                           | 46               | 2799.00877                | bb367a3240...20b40e528b, b345934c17...4d6511b20b, 91401271d3...b3c4275114, 59c0081328...c04d597c06                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| 000000000000000000010000000000000000000000000000000000000bffffff | 53               | 1438.83780     | 76258.40364   | 5                 | 5                           | 48               | 1438.83780                | f56d9c29a0...ba7908a112, e6a1f584b4...6c2b450f80, d55bde4cbd...ca6e5a4f8b, a811cfb07c...3bc4d96b17, 32ed3dcd02...9a4bcb9d05                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
| 0000000000000000000100000000000000000000000000000000000020ffffff | 36               | 4015.53315     | 144559.19363  | 4                 | 4                           | 32               | 4015.53315                | c60c12298c...cc41aea519, b89df7002f...1f7f073105, 6f8038d51b...ee34147b92, 31ff7db5d5...65199f1d14                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |

```

3.

```
DEBUG[2024-02-01 18:49:36.198] [..hStart/metachain] [metachain/1269/18448096/(BLOCK)] Final selected nodes from auction list
+------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------+
| Owner                                                            | Registered key                                                                                                                                                                                   | Qualified TopUp per node |
+------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------------------+
| 0000000000000000000100000000000000000000000000000000000061ffffff | 90e951e3342ec776a0a73272c88b75bda92dc7dca5eb38528b1c6acb81c7ba0f66430b5cbc3cdeacbfc378a4f0584c10ba8d96b9d789c21a0aad7d15263e2b52a2e5ed597268d414735a50c07cde3ec1952b9fdb3ac3dcd95bfa2649f006a605 | 29553.89004              |
| 0000000000000000000100000000000000000000000000000000000061ffffff | b3a052a7bb2cc302f6f5189e31d59e6603e151eaa6a1785258615b259156cf37907e95e8077f4d3ae2aaffb76510d810dc81c318542ffaec543c52bd70950b285140ac908b162f9be92a96f8bf38d21212db5c07e6cf11f23dac6ee71007ee93 | 29553.89004              |
| 0000000000000000000100000000000000000000000000000000000061ffffff | d7f14109f8276376e77e714f4aef1cb68bd0ad3ef3a07a9b925627a386e4b4e134d74eebfcba11a359571dddd223db0849ebbc7687a1d497029da99fec36a89c75dc65dd5b851d1e8424fe4dd6cfbf90954b27973e7d0529952ccda649faaf0d | 29553.89004              |

```


## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: epochStart/metachain/auctionListDisplayer.go [hunk 1]
```diff
@@ -9,56 +9,61 @@ import (
...
 
-	for idx, validator := range list {
-		pubKeys += getShortKey(validator.GetPublicKey())
-		addDelimiter := idx != len(list)-1
-		if addDelimiter {
-			pubKeys += ", "
+	lines := make([]*display.LineData, 0, len(ownersData))
+	for ownerPubKey, owner := range ownersData {
+		line := []string{
+			hex.EncodeToString([]byte(ownerPubKey)),
```
Comment (raduchis) [inline epochStart/metachain/auctionListDisplayer.go:55]: don't we want this bech32?

Comment (iulianpascalau) [reply epochStart/metachain/auctionListDisplayer.go:55]: Agree. Please do this properly by injecting 2 address converters on the constructor of the auctionListDiplayer

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
For the above comments

File: epochStart/metachain/auctionListSelector.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ import (
 	"github.com/multiversx/mx-chain-go/state"
 )
 
-type ownerAuctionData struct {
+type OwnerAuctionData struct {
```
[the remark this answers is not in the document]
Comment (mariusmihaic) [reply epochStart/metachain/auctionListSelector.go:18]: Fixed

File: epochStart/metachain/auctionListDisplayer.go (hunk 1 above)
Comment (mariusmihaic) [reply epochStart/metachain/auctionListDisplayer.go:55]: Yes, but was thinking about doing it on another PR.
Will do it here

**Review by iulianpascalau — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Pull request merged**